app/analytics/[questId]/page.tsx

```diff
@@ -0,0 +1,390 @@
+"use client";
+
+import React, { useCallback, useEffect, useState } from "react";
+import styles from "@styles/questboost.module.css";
+import analyticsStyles from "@styles/analytics.module.css";
+import { useRouter } from "next/navigation";
+import BackButton from "@components/UI/backButton";
+import {
+  AreaChart,
+  Area,
+  XAxis,
+  YAxis,
+  Tooltip,
+  ResponsiveContainer,
+  CartesianGrid,
+} from "recharts";
+import {
+  getQuestActivityData,
+  getQuestById,
+  getQuestParticipants,
+  getQuestsParticipation,
+  getUniqueVisitorCount,
+} from "@services/apiService";
+import { getMonthName } from "@utils/stringService";
+import { QuestDocument } from "../../../types/backTypes";
+import { numberWithCommas } from "@utils/numberService";
+import { CDNImg } from "@components/cdn/image";
+import { useMediaQuery } from "@mui/material";
+import AnalyticsSkeleton from "@components/skeletons/analyticsSkeleton";
+
+type BoostQuestPageProps = {
+  params: {
+    questId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const router = useRouter();
+
+  const { questId } = params;
+  const [loading, setLoading] = useState<boolean>(true);
+  const [graphData, setGraphData] = useState([]);
+  const [questParticipationData, setQuestParticipationData] = useState([]);
+  const [questParticipants, setQuestParticipants] = useState(0);
+  const [uniqueVisitors, setUniqueVisitors] = useState(0);
+  const isMobile = useMediaQuery("(max-width:768px)");
+  const [questData, setQuestData] = useState<QuestDocument>({
+    id: 0,
+    name: "loading",
+    desc: "loading",
+    issuer: "loading",
+    category: "loading",
+    rewards_endpoint: "",
+    logo: "",
+    rewards_img: "",
+    rewards_title: "loading",
+    rewards_nfts: [],
+    img_card: "",
+    title_card: "",
+    hidden: false,
+    disabled: false,
+    expiry_timestamp: "loading",
+    mandatory_domain: null,
+    expired: false,
+    rewards_description: null,
+    additional_desc: null,
+  });
```

Comment:
to initialize in another file maybe ? wdyt ?

Reply:
done